cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -120,13 +120,21 @@ func newEtcdClient() (client *etcd.Client) {
 			break
 		}
 	}
-	client = etcd.NewClient([]string{*etcd_server})
-	if client == nil {
-		return nil
+	timeout := 10 * time.Second
+	// loop until we have > 0 machines && machines[0] != ""
+	for start := time.Now(); time.Now().Sub(start) < timeout; time.Sleep(time.Second) {
```

Comment:
This is the timeout loop syntax that we use in all of our e2e test code and it's nice to have the way that we write the loop consistent across the codebase. See #4292 for background. 

I'm not opposed to changing it here, I just want there to be a good reason and think about whether that applies more widely than this one PR.